### DISCUSSION TITLE
[.net] Comparison of features with Python version

### DISCUSSION THREAD

**Discussion by ChristianWeyer** [microsoft/autogen#2787] [General]
Do we have a comparison of all features that the .NET and Python version implement?

**Comment by LittleLittleCloud** [comment]
Hey @ChristianWeyer i will add a page about feature comparison in autogen net website 

The TLDR is autogen.net supports nearly all agent conversation pattern like single agent chat, two agent chat, group chat, graph chat. What it doesn’t support is specialized agent like rag agent or web surfer agent, and nearly all features under /contrib because of the ecosystem difference between python and dotnet.

In general, you can use autogen.net to construct the same agentic workflow pattern in python autogen, but you might not be able to achieve the same functionality of an agent because some support libraries are missing in dotnet

**Reply by ChristianWeyer** [reply]
Thx man!

**Reply by ChristianWeyer** [reply]
> you might not be able to achieve the same functionality of an agent because some support libraries are missing in dotnet

Could you elaborate on this, including examples?

**Reply by LittleLittleCloud** [reply]
Surr, for example if a pdf rag agent in python autogen uses a python package to process pdfs and in dotnet there is no corresponding dotnet package that provides the same functionality to process pdf…

**Reply by ChristianWeyer** [reply]
> Hey @ChristianWeyer i will add a page about feature comparison in autogen net website
> 
> The TLDR is autogen.net supports nearly all agent conversation pattern like single agent chat, two agent chat, group chat, graph chat. What it doesn’t support is specialized agent like rag agent or web surfer agent, and nearly all features under /contrib because of the ecosystem difference between python and dotnet.
> 
> In general, you can use autogen.net to construct the same agentic workflow pattern in python autogen, but you might not be able to achieve the same functionality of an agent because some support libraries are missing in dotnet

Hi! Did you already find some time to create this page with the comparison?
Thanks!

**Reply by LittleLittleCloud** [reply]
Sorry,  I totally forget it. Just create a tracking issue in case I forget it again

**Reply by LittleLittleCloud** [reply]
#3184
